helm-branches.yaml

```diff
@@ -1,35 +1,51 @@
 deploy:quality:helm:branches:
   extends: .helm:deploy
   environment:
-    name: ${CI_COMMIT_REF_SLUG}
-    url: https://${DOMAIN}
+    name: review/$CI_BUILD_REF_NAME
+    url: https://${CI_COMMIT_REF_SLUG}.${DOMAIN_QUALITY}
     on_stop: stop:quality
   variables:
     CI_COMMIT_TAG: ${CI_COMMIT_SHORT_SHA}
     GOOGLE_KEY: ${GOOGLE_KEY_QUALITY}
     CLUSTER_NAME: ${CLUSTER_NAME_QUALITY}
     CLUSTER_ZONE: ${CLUSTER_ZONE_QUALITY}
-    SECRET_YAML: ${SECRET_YAML_QUALITY}
+    #SECRET_YAML: ${SECRET_YAML_FEATURE}
     NAMESPACE: ${NAMESPACE_QUALITY}
     ENVIRONMENT: feature
     APP_NAME: ${CI_COMMIT_REF_SLUG}
   before_script:
     - |
+      # use a specific CI/CD variable if SECRET_BRANCH is specified in gitlab-ci
+      if [ -n "${SECRET_BRANCH}" ]; then
+        export SECRET_YAML=${!SECRET_BRANCH}
```

Comment:
do i take it right that the "place" where the secret is stored is written inside `SECRET_BRANCH`? that would go into ci variables, right?
what is the reason for picking this pattern?

Reply:
Well, if you want a branch to have a specific secfret different from QA, this way you can set is as a gitlab ci variable called SECRET_BRANCH. Initially it might be the same as the quality one, but you can customize a secret (and specify a different name as well) and mount it in the helm/feature/values.yaml in place of the qualty one.

Reply:
Is this idea written in the readme?

Reply:
also I did get it wrong, i.e. SECRET_BRANCH contains the YAML not the "place" where the yaml is stored, right? if this is the case I'd call it SECRET_YAML_BRANCH to keep pattern of others